torch/utils/_mode_utils.py

```diff
@@ -0,0 +1,123 @@
+import functools
+from typing import Iterator
+from dataclasses import dataclass
+
```

Comment:
would be nice to have a top level comment for the file

Reply:
Added!